src/inmanta/data/dataview.py

```diff
@@ -1271,27 +1272,39 @@ def __init__(
             prelude=None,
         )
         self._prelude_query_builder = prelude_query_builder
+        self._prelude_query_builder_extra = prelude_query_builder_extra
 
     @property
     def offset(self) -> int:
         """The current offset of the values to be used for filter statements"""
-        return len(self.values) + len(self._prelude_query_builder.values) + 1
+        extra_values_count = sum(len(builder.values) for builder in self._prelude_query_builder_extra)
+        return len(self.values) + len(self._prelude_query_builder.values) + extra_values_count + 1
 
     def build(self) -> tuple[str, list[object]]:
         prelude_query, prelude_values = self._prelude_query_builder.build()
-        prelude_query_in_with_block = f"WITH prelude AS ({prelude_query})"
-        delegate: SimpleQueryBuilder = SimpleQueryBuilder(
+        prelude_section = f"prelude AS ({prelude_query})"
+        extra_prelude_values = []
+        extra_prelude_queries = []
+
+        for idx, extra_builder in enumerate(self._prelude_query_builder_extra):
+            extra_query, extra_values = extra_builder.build()
+            prelude_name = f"extra_prelude_{idx + 1}"
```

Comment:
Indeed, I'll change it: instead of a list of preludes the argument will become a list of tuples where the first element of each tuple is the name of the table and the second argument is the prelude